src/cli/dcos_docker/commands/create.py

```diff
@@ -78,6 +78,56 @@ def _validate_docker_network(
         raise click.BadParameter(message=message)
 
 
+def _validate_port_map(
+    ctx: click.core.Context,
+    param: Union[click.core.Option, click.core.Parameter],
+    value: Any,
+) -> Dict[str, Any]:
+    """
+    Turn port map strings into a Dict that ``docker-py`` can use.
+    """
+    # We "use" variables to satisfy linting tools.
+    for _ in (ctx, param):
+        pass
+
+    ports = {}
+    for ports_definition in value:
+        parts = ports_definition.split(':')
+
+        # TODO(jieyu): Consider support the full docker syntax.
+        # https://docs.docker.com/engine/reference/run/#expose-incoming-ports
+        if len(parts) != 2:
+            message = (
+                '"{ports_definition}" is not a valid port map. '
+                'Please follow this syntax: <hostPort>:<containerPort>'
+            ).format(ports_definition=ports_definition)
+            raise click.BadParameter(message=message)
+
+        hostPort, containerPort = parts
+        if not hostPort.isdigit():
+            message = (
+                '"{hostPort}" is not a number.'
+            ).format(hostPort=hostPort)
+            raise click.BadParameter(message=message)
+        if not containerPort.isdigit():
+            message = (
+                '"{containerPort}" is not a number.'
+            ).format(containerPort=containerPort)
+            raise click.BadParameter(message=message)
+        if int(hostPort) < 0 or int(hostPort) > 65535:
+            message = (
+                '"{hostPort}" is not a valid port number.'
+            ).format(hostPort=hostPort)
+            raise click.BadParameter(message=message)
+        if int(containerPort) < 0 or int(containerPort) > 65535:
+            message = (
+                '"{containerPort}" is not a valid port number.'
+            ).format(containerPort=containerPort)
+
+        ports[containerPort + '/tcp'] = int(hostPort)
```

Comment:
❓ Do you think that it is worth mentioning in the CLI option help that the protocol is tcp?

Reply:
If the user specifies multiple values, and >= 2 have the same container port, there will be no error, but only the last value will be used. I think that there should be an error.

Reply:
Done.